src/MatchResult/MatchResultInterface.php

```diff
@@ -20,4 +22,12 @@
      * @return bool
      */
     public function isSuccess();
+
+    /**
+     * Modifies the response to properly indicate the kind of failure.
+     *
+     * @param  ResponseInterface $response
+     * @return ResponseInterface
+     */
+    public function modifyResponse(ResponseInterface $response);
```

Comment:
I think you mean AbstractFailedRequest? Theoretically yes, but then it's not part of the contract which is returned by the match method and we have to fall back to ductyping/instanceof checks. If at all, it should be its own interface.